### PR TITLE
Adding new Getting Started content

### PR DESCRIPTION
**Pull request by thesmithee**
Changing DNX to CLI tooling. The content is pretty much the same.

After this is merged, we should _seriously consider_ leaving this kind of getting started in one place, ofr instance the Core website (http://dotnet.github.io/getting-started/). 

/cc @cartermp @BethMassi @richlander

### CODE REVIEW

File: docs/getting-started/installing/installing-core-linux.md [hunk 1]
```diff
@@ -2,119 +2,71 @@
 
 By [Zlatko Knezevic](https://github.com/blackdwarf)
 
-These instructions will lead you through acquiring the .NET Core DNX SDK via the [.NET Version Manager (DNVM)](https://github.com/aspnet/dnvm) and running a “Hello World” demo on Linux.
-
-.NET Core NuGet packages and the .NET Core DNX SDKs are available on the [ASP.NET ‘vnext’ myget feed](https://www.myget.org/F/aspnetvnext), which you can more easily view on [gallery](https://www.myget.org/gallery/aspnetvnext) for the feed.
+This document will lead you through acquiring the .NET Core and its associated CLI toolchainand running a “Hello World” demo on Linux.
```
Comment (gkhanna79) [inline docs/getting-started/installing/installing-core-linux.md]: "toolchain and"

File: docs/getting-started/installing/installing-core-linux.md [hunk 2]
```diff
@@ -2,119 +2,71 @@
...
-
-### Installing the required packages
-
-Install the `libunwind8`, `libssl-dev` and `unzip` packages:
-
-```console
-sudo apt-get install libunwind8 libssl-dev unzip
-
-```
+These instructions have been written and tested on Ubuntu 14.04 LTS, since that is the main Linux distribution the .NET Core team uses.  We are always accepting new pull requests on [our GitHub repo](https://www.github.com/dotnet/coreclr/) that address running on other Linux distributions. The only requirement is that they do not break the ability to use Ubuntu 14.04 LTS.
```
Comment (gkhanna79) [inline docs/getting-started/installing/installing-core-linux.md]: We should be validating these on all platforms we support - Ubuntu, OSX, Windows.

File: docs/getting-started/installing/installing-core-linux.md [hunk 3]
```diff
@@ -2,119 +2,71 @@
...
 
-### Certificates
+### Setting up the apt-get feed
 
-You need to import trusted root certificates in order to restore NuGet packages. You can do that with the `mozroots` tool.
+We will use apt-get, the native Ubuntu package installer, to install the .NET Core SDK. In order to get the package, however, we will need to add a new package feed. The below commands will do this.
```
Comment (gkhanna79) [inline docs/getting-started/installing/installing-core-linux.md]: We should exemplify the experience for all supported OS platforms and not just Ubuntu.

File: docs/getting-started/installing/installing-core-linux.md [hunk 4]
```diff
@@ -2,119 +2,71 @@
...
-
-See the asterisk in the listing above? It’s purpose is to tell you which runtime is now active. “Active” here means that all of the interaction with your projects and .NET Core will use this runtime.
+This will install the package and all of the dependencies. It will also add the toolchain to your $PATH, so they will be available the next time you drop down into the terminal.  
 
 That’s it! You now have the .NET Core runtime installed on your machine and it is time to take it for a spin.
 
 ## Write your App
 
-his being an introduction-level document, it seems fitting to start with a “Hello World” app. Here’s a very simple one you can copy and paste into a CS file in a directory.
+This being an introduction-level document, it seems fitting to start with a “Hello World” app. Here’s a very simple one you can copy and paste into a CS file in a directory.
```
Comment (gkhanna79) [inline docs/getting-started/installing/installing-core-linux.md]: Why are we not exemplifying using "dotnet new"? Shouldn't we strive for guidance consistency here?

File: docs/getting-started/installing/installing-core-linux.md [hunk 5]
```diff
@@ -2,119 +2,71 @@
...
 
 ```
 
-A more ambitious example is available on the [corefxlab repo](https://www.github.com/dotnet/corefxlab/) that will print out a pretty picture based on the argument you provide at runtime. If you wish to use this example, simply save the [C# file](https://raw.githubusercontent.com/dotnet/corefxlab/master/demos/CoreClrConsoleApplications/HelloWorld/HelloWorld.cs) to a directory somewhere on your machine.
-
 The next thing you will need is a `project.json` file that will outline the dependencies of an app, so you can **actually** run it. Use the contents below, it will work for both examples above. Save this file in a directory next to the CS file that contains your code.
 
 ```json
-{
+ {
```
Comment (gkhanna79) [inline docs/getting-started/installing/installing-core-linux.md]: See above.

File: docs/getting-started/installing/installing-core-linux.md [hunk 6]
```diff
@@ -122,17 +74,50 @@ The next thing you will need is a `project.json` file that will outline the depe
 
 ## Run your App
 
-You need to restore packages for your app, based on your project.json, with `dnu restore`.
+You need to restore packages for your app, based on your project.json, with `dotnet restore`.
```
Comment (gkhanna79) [inline docs/getting-started/installing/installing-core-linux.md]: I would suggest extending the sentence above with something like:

> ...and execute the app using dotnet run

to account for the two commands you have specified below.

File: docs/getting-started/installing/installing-core-linux.md [hunk 7]
```diff
@@ -122,17 +74,50 @@ The next thing you will need is a `project.json` file that will outline the depe
...
+
+```
+
+## Compile your application
 
+Running from source is great for rapid prototyping and trying out things. However, in due time you will want to actually compile your application to get increase in speed and similar benefits. In order to do that, we will use `dotnet compile` command that will produce a runnable executable for our Hello World app.
+
+While you're still in the application's directory, type
+
+    dotnet compile
```
Comment (gkhanna79) [inline docs/getting-started/installing/installing-core-linux.md]: Is it now going to be "dotnet build"?

File: docs/getting-started/installing/installing-core-linux.md [hunk 8]
```diff
@@ -122,17 +74,50 @@ The next thing you will need is a `project.json` file that will outline the depe
...
+    
+This will produce a `bin` directory in your directory. The structure of the drop path is `./bin/[configuration]/[framework]/`. Configuration refers to either *Release* or *Debug*, while framework is essentially a framework ID (i.e. dnxcore50). Inside this directory there will be several files, the most important of which is the binary that will have the same name as your application. Running this will give us the message we saw in the previous example. 
+
+**Note**: this binary requires a shared runtime to be installed on the machine. If you wish to create a self-contained application that includes the runtime and that you can just copy over to another machine, you will need to use `dotnet publish` command or compile your application
```
Comment (gkhanna79) [inline docs/getting-started/installing/installing-core-linux.md]: Is there going to be documentation for Shared runtimes, servicing runtimes, etc? We should consider linking it from here to avoid the reader the headache of finding it and understanding the semantics.

File: docs/getting-started/installing/installing-core-linux.md [hunk 9]
```diff
@@ -122,17 +74,50 @@ The next thing you will need is a `project.json` file that will outline the depe
...
 
-## Building .NET Core from source
+After the compile command finishes, we can just run the resulting binary. By convention, the compile command drops the results in ./bin/[configuration]/[framework]/native/[binary name]. Running this binary will get us our greeting! 
+
+> **Note**
+> This capability is still in its infancy. Therefore, only the simplest of programs will be able to be natively compiled. 
+
+## Building .NET Core runtime from source
+
+.NET Core is an open source project that is hosted on GitHub. This means that you can, at any given time, clone the repository and build .NET Core from source. This is a more advanced scenario that is usually used when you want to add features to the .NET runtime or the BCL or if you are a contributor to these projects. The detailed instruction on how to build .NET Core windows can be found in the [Build .NET Core on Linux](https://github.com/dotnet/coreclr/blob/master/Documentation/building/linux-instructions.md) on GitHub.
```
Comment (gkhanna79) [inline docs/getting-started/installing/installing-core-linux.md]: .NET Core is comprised of multiple repos - CoreCLR, CoreFX, Roslyn, CoreRT, etc. The above phrasing should be updated to help explain the right concepts to our development community. It will be great if we can link to a PPT/Article that explains what .NET Core is.

File: docs/getting-started/installing/installing-core-osx.md [hunk 10]
```diff
@@ -2,84 +2,11 @@
 
 By [Zlatko Knezevic](https://github.com/blackdwarf)
 
-These instructions will lead you through acquiring the .NET Core DNX SDK via the [.NET Version Manager (DNVM)](https://github.com/aspnet/dnvm) and running a “Hello World” demo on OS X.
+This document will lead you through acquiring the .NET Core and its associated CLI toolchainand running a “Hello World” demo on OS X. The packages are set to work on OS X version 10.10.
```
Comment (gkhanna79) [inline docs/getting-started/installing/installing-core-osx.md]: Should we avoid the duplication of the text?

File: docs/getting-started/installing/installing-core-linux.md (hunk 2 above)
Comment (blackdwarf) [inline docs/getting-started/installing/installing-core-linux.md]: Don't really understand the comment?

File: docs/getting-started/installing/installing-core-linux.md (hunk 4 above)
Comment (blackdwarf) [inline docs/getting-started/installing/installing-core-linux.md]: Yeah, this is a short fix, I'm hoping that we can replace these documents with the pointer to the "getting started" on the Core website.

File: docs/getting-started/installing/installing-core-linux.md (hunk 3 above)
Comment (blackdwarf) [inline docs/getting-started/installing/installing-core-linux.md]: Well, other files have other platforms, so they are talking about those specifically. :)

File: docs/getting-started/installing/installing-core-linux.md [hunk 11]
```diff
@@ -122,17 +74,50 @@ The next thing you will need is a `project.json` file that will outline the depe
 
 ## Run your App
 
-You need to restore packages for your app, based on your project.json, with `dnu restore`.
+You need to restore packages for your app, based on your project.json, with `dotnet restore`.
 
 ```console
-dnu restore
-dnx run
+dotnet restore
```
Comment (cartermp) [inline docs/getting-started/installing/installing-core-linux.md]: Nit: 

Adding "$" (or other common symbol) makes it a bit more clear that this is exactly what you have to type in the command line.

File: docs/getting-started/installing/installing-core-linux.md [hunk 12]
```diff
@@ -122,17 +74,50 @@ The next thing you will need is a `project.json` file that will outline the depe
...
+
+## Create a single native binary 
+
+Finally, let's exercise a new feature that we've added to our .NET Command Line Interface: producing single native binaries. These binaries do not require a shared runtime to work; you can just copy the single file over to another Ubuntu machine and just run it. 
+
+The process is pretty similar to the above, with the addition of one more switch. 
+
+```console
+dotnet restore
+dotnet compile --native
```
Comment (cartermp) [inline docs/getting-started/installing/installing-core-linux.md]: Same as above

File: docs/getting-started/installing/installing-core-linux.md (hunk 9 above)
Comment (cartermp) [inline docs/getting-started/installing/installing-core-linux.md]: +1, this could stand to have some more clarity on what the components are before linking to the building instructions.  If by "building .NET Core from source" actually means building coreclr from source (at least that's where the link leads), it should state that briefly.

What is ".NET Core windows"?

File: docs/getting-started/installing/installing-core-osx.md [hunk 13]
```diff
@@ -93,54 +20,66 @@ public class Program
...
+
+**Note**: this binary requires a shared runtime to be installed on the machine. If you wish to create a self-contained application that includes the runtime and that you can just copy over to another machine, you will need to use `dotnet publish` command or compile your application 
+
+## Building .NET Core runtime from source
+
+.NET Core is an open source project that is hosted on GitHub. This means that you can, at any given time, clone the repository and build .NET Core from source. This is a more advanced scenario that is usually used when you want to add features to the .NET runtime or the BCL or if you are a contributor to these projects. The detailed instruction on how to build .NET Core windows can be found in the [Build .NET Core on Linux](https://github.com/dotnet/coreclr/blob/master/Documentation/building/osx-instructions.md) on GitHub.
```
Comment (cartermp) [inline docs/getting-started/installing/installing-core-osx.md]: [Build .NET Core on Linux] ---> [Build .NET Core on OS X]

**Comment by cartermp** [comment]
LGTM modulo feedback.

I agree about there being a more central place for these instructions.

**Comment by blackdwarf** [comment]
Ups, I didn't see these comments at all. GH and their notifications. I will modify and update soon-ish.

File: docs/getting-started/installing/installing-core-linux.md (hunk 2 above)
Comment (blackdwarf) [inline docs/getting-started/installing/installing-core-linux.md]: @gkhanna79 ah, now I see what you meant. This is a doc just for Ubuntu, so that's why this paragraph sits here. I think I will rename it to be "Installing on Ubuntu" and remove this para.

File: docs/getting-started/installing/installing-core-linux.md (hunk 7 above)
Comment (blackdwarf) [inline docs/getting-started/installing/installing-core-linux.md]: Yup.

File: docs/getting-started/installing/installing-core-linux.md (hunk 8 above)
Comment (blackdwarf) [inline docs/getting-started/installing/installing-core-linux.md]: @gkhanna79 good points, we will definitely add these docs in the future.

File: docs/getting-started/installing/installing-core-linux.md (hunk 11 above)
Comment (blackdwarf) [inline docs/getting-started/installing/installing-core-linux.md]: I actually hate that, because people then cannot easily copy-paste.

File: docs/getting-started/installing/installing-core-linux.md (hunk 9 above)
Comment (blackdwarf) [inline docs/getting-started/installing/installing-core-linux.md]: I removed this part, not really part of the getting started docs.

File: docs/getting-started/installing/installing-core-osx.md (hunk 10 above)
Comment (blackdwarf) [inline docs/getting-started/installing/installing-core-osx.md]: @gkhanna79 this makes sense, but will have to do it in a separate PR.
